File: src/Messages/Device/StatePower_Device.java
package Messages.Device;

import LifxCommander.CommonMethods;
import Messages.DataTypes.Payload;

public class StatePower_Device extends Payload {
	int code = 22;
	int level;				// 16-Bits (Unsigned)
	
	public StatePower_Device() {
		level = Values.Power.OFF;
	}
	
	public StatePower_Device(int level) {
		this.level = level;
	}
	
	public StatePower_Device(StatePower_Device statePower) {
		level = statePower.level;
	}
	
	public int getCode() {
		return code;
	}
	
	public int getLevel() {
		return level;
	}
	
	public void setLevel(int level) {
		this.level = level;
	}
	
	public void setFromCommandByteArray(byte[] byteArray) {
		String levelBinStr = CommonMethods.convertByteToBinaryString(byteArray[37]).concat(CommonMethods.convertByteToBinaryString(byteArray[36]));
		level = Integer.parseInt(levelBinStr, 2);
	}
	
}

File: src/main/java/org/stevenlowes/tools/lifxcontroller/commands/response/device/StatePowerDevice.kt
package org.stevenlowes.tools.lifxcontroller.commands.response.device

import org.stevenlowes.tools.lifxcontroller.Utils
import org.stevenlowes.tools.lifxcontroller.commands.response.ResponsePayload
import org.stevenlowes.tools.lifxcontroller.values.Level

data class StatePowerDevice(val level: Level = Level.MIN) : ResponsePayload(22) {

    companion object {
        fun loadFrom(byteArray: ByteArray): StatePowerDevice {
            val levelBinStr = Utils.convertByteToBinaryString(byteArray[37]) + Utils.convertByteToBinaryString(
                    byteArray[36])
            val level = Level(Integer.parseInt(levelBinStr, 2))

            return StatePowerDevice(level)
        }
    }
}